test/test_llm_connect.py

```diff
@@ -121,4 +121,5 @@ def test_azure():
         base=os.getenv("AZURE_TEST_OPENAI_API_BASE"),
     )
 
-    assert convo.set_api_key(os.getenv("AZURE_TEST_OPENAI_API_KEY"))
```

Comment:
Is this a workaround for the fact that we don't have that available on the GitHub runner? I would prefer to do this with a conditional skip or better with mocking. This way, the meaning of the test changes.

Reply:
Yes, this is a wrokaround, has been reverted back

Reply:
To make tests run successfully, besides OPENAI_API_KEY, we need to add the following env to github secrets:
AZURE_TEST_OPENAI_MODEL_NAME=
AZURE_TEST_OPENAI_DEPLOYMENT_NAME=
AZURE_TEST_OPENAI_API_VERSION=
AZURE_TEST_OPENAI_API_BASE=
AZURE_TEST_OPENAI_API_KEY=